File: app/src/main/java/com/cookplan/providers/ToDoListProvider.java
package com.cookplan.providers;

import com.cookplan.models.ToDoCategory;
import com.cookplan.models.ToDoItem;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;
import io.reactivex.Single;

/**
 * Created by DariaEfimova on 24.04.17.
 */

public interface ToDoListProvider {

    Observable<List<ToDoCategory>> getUserToDoCategoriesList();

    Observable<List<ToDoItem>> getUserToDoList();

    Single<ToDoItem> createToDoItem(ToDoItem item);

    Single<ToDoCategory> createToDoCategory(ToDoCategory category);

    Single<ToDoItem> updateToDoItem(ToDoItem item);

    Completable removeToDoItem(ToDoItem item);

    Completable removeToDoCategory(ToDoCategory category);
}

File: app/src/main/java/com/cookplan/providers/ToDoListProvider.kt
package com.cookplan.providers

import com.cookplan.models.ToDoCategory
import com.cookplan.models.ToDoItem

import io.reactivex.Completable
import io.reactivex.Observable
import io.reactivex.Single

/**
 * Created by DariaEfimova on 24.04.17.
 */

interface ToDoListProvider {

    fun getUserToDoCategoriesList(): Observable<List<ToDoCategory>>

    fun getUserToDoList(): Observable<List<ToDoItem>>

    fun getCompanyToDoList(companyId: String): Observable<List<ToDoItem>>

    fun createToDoItem(item: ToDoItem): Single<ToDoItem>

    fun createToDoCategory(category: ToDoCategory): Single<ToDoCategory>

    fun updateToDoItem(item: ToDoItem): Single<ToDoItem>

    fun removeToDoItem(item: ToDoItem): Completable

    fun removeToDoCategory(category: ToDoCategory): Completable
}
